packages/flutter/test/material/stepper_test.dart

```diff
@@ -953,4 +954,100 @@ void main() {
 
      expect(material.elevation, 2.0);
    });
+
+  testWidgets('Stepper horizontal preserves state', (WidgetTester tester) async {
+    Widget buildFrame() {
+      return const MaterialApp(
+        home: Scaffold(
+          body: Center(
+            // Must break this out into its own widget purely to be able to call `setState()`
```

Comment:
Would a StatefulBuilder help here? You can capture the StateSetter in the builder.